bindings/rust/evmc-declare-tests/src/lib.rs

```diff
@@ -0,0 +1,22 @@
+use evmc_vm::EvmcVm;
+use evmc_vm::ExecutionResult;
+use evmc_vm::ExecutionContext;
+#[macro_use]
+use evmc_declare::evmc_declare_vm;
+
+#[evmc_declare_vm(FooVm = "ewasm")]
+pub struct FooVM {
+    a: i32,
+}
+
+impl EvmcVm for FooVM {
+    fn init() -> Self {
+        FooVM { 
+            a: 105023,
+        }
+    }
+
+    fn execute(&self, code: &[u8], context: &ExecutionContext) -> ExecutionResult {
```

Comment:
As a next step perhaps we could use the `Result` pattern, where `Error` is translated into a failure code in `ExecutionResult`. But I wouldn't touch that in this PR.

Reply:
Yeah, later

Reply:
Created issue #309 to track this.